### PR TITLE
[v6r19] GLUE2_2_CS

### PR DESCRIPTION
**Pull request by andresailer**
Some development for #3633 

~~This should be enough to at least discover new CEs at CERN.~~
This should be enough to discover most CEs: HTCondorCE at CERN, ARC and CREAM
The queue names were tough to obtain from the GLUE2 schema, so they slightly differ in some cases.

~~Except for MaxRam I have not found any of the information needed, architecture always comes out as amd64, OS as linux and nothing else... (see https://github.com/DIRACGrid/DIRAC/issues/3633#issuecomment-380792221)~~

Some information is only published for some CEs...

BEGINRELEASENOTES

*Configuration

NEW: BDII2CSAgent: new options:
   * GLUE2URLs, if set this is queried in addition to the other BDII, 
   * GLUE2Only to turn off looking on the old schema, if **true** only the main BDII URL is queried
   * Host to set the BDII host to search

NEW: dirac-admin-add-resources new option G/glue2 , enable looking at GLUE2 Schema, H/host to set the host URL to something else

ENDRELEASENOTES

TODO:
- [x] add doc for new options
- [x] look for ways to get more of the information

### CODE REVIEW

**Comment by andresailer** [comment]
I reworked things to start by looking up the `Policies` allowing the given VO and getting the `ComputingShare`s associated with the policy.
Added parsing of options of CREAM and ARC CEs as well.

**Comment by andresailer** [comment]
Please see updated first comment,
@fstagni @VladimirRomanovsky @atsareg Please review

File: Core/Utilities/Glue2.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
+""" Module collecting functions dealing with the GLUE2 information schema
+
+:author: A.Sailer
+
+"""
+
+from pprint import pformat
+
+from DIRAC import gLogger
+from DIRAC.Core.Utilities.ReturnValues import S_OK, S_ERROR
```
Comment (fstagni) [inline Core/Utilities/Glue2.py]: just `from DIRAC`?

File: ConfigurationSystem/Agent/Bdii2CSAgent.py [hunk 2]
```diff
@@ -177,19 +184,30 @@ def __getBdiiCEInfo( self, vo ):
...
+      resultAlt = getBdiiCEInfo(vo, host=bdii, glue2=self.glue2Only)
       if resultAlt['OK']:
         totalResult['Value'].update( resultAlt['Value'] )
       else:
         self.log.error( "Failed getting information from %s " % bdii, resultAlt['Message'] )
         message = ( message + "\n" + resultAlt['Message'] ).strip()
 
+    for glue2URL in self.glue2URLs:
+      if self.glue2Only:
+        break
```
[the remark this answers is not in the document]
Comment (andresailer) [reply ConfigurationSystem/Agent/Bdii2CSAgent.py]: If `self.glue2Only` then we use the main BDII URL, so we don't check the glue2URLs.

https://github.com/DIRACGrid/DIRAC/pull/3661/files#diff-47c2980d755ffe3f5051e7e5db462761R187

This basically accounts for the eventual transition to GLUE2 only.

File: Core/Utilities/Glue2.py (hunk 1 above)
Comment (andresailer) [reply Core/Utilities/Glue2.py]: I copied the import from Grid.py, f234de5d72 , I thought there was a reason for using the actual location for the import, so I kept it

File: ConfigurationSystem/Agent/Bdii2CSAgent.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fstagni) [reply ConfigurationSystem/Agent/Bdii2CSAgent.py]: So, basically `glue2Only` flag means: "get info ONLY from `self.host` and ONLY from the glue2 schema in there". Correct?

File: Core/Utilities/Glue2.py (hunk 1 above)
Comment (fstagni) [reply Core/Utilities/Glue2.py]: I don't remember any specific reason.

**Comment by fstagni** [comment]
Travis is failing because of installing fts3 package with pip. I corrected it in https://gitlab.cern.ch/fts/fts-rest/merge_requests/15
It's basically due to the upgrade of pip.

File: ConfigurationSystem/Agent/Bdii2CSAgent.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (andresailer) [reply ConfigurationSystem/Agent/Bdii2CSAgent.py]: Yes, I updated the documentation to be clear on this.

**Pull request merged**